adp-algorithms/util/src/main/java/de/metanome/util/InclusionDependencyUtil.java

```diff
@@ -0,0 +1,86 @@
+package de.metanome.util;
+
+import com.google.common.base.Preconditions;
+import de.metanome.algorithm_integration.ColumnIdentifier;
+import de.metanome.algorithm_integration.results.InclusionDependency;
+import it.unimi.dsi.fastutil.ints.Int2IntMap;
+import it.unimi.dsi.fastutil.ints.Int2IntOpenHashMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class InclusionDependencyUtil {
+
+  public List<InclusionDependency> getMax(final Collection<InclusionDependency> toMinify) {
+    final List<InclusionDependency> ind = new LinkedList<>(toMinify);
+    ind.sort(Comparator.comparing(this::degreeOf).reversed());
+
+    final List<InclusionDependency> maxInd = new ArrayList<>();
+    while (!ind.isEmpty()) {
+      final InclusionDependency current = ind.remove(0);
+      ind.removeAll(containedOf(current, ind));
+      maxInd.add(current);
+    }
+
+    return maxInd;
+  }
+
+  private int degreeOf(final InclusionDependency ind) {
+    return ind.getDependant().getColumnIdentifiers().size();
+  }
+
+  private List<InclusionDependency> containedOf(final InclusionDependency ind,
+      final List<InclusionDependency> candidates) {
+
+    final List<InclusionDependency> contained = new ArrayList<>();
+    for (final InclusionDependency candidate : candidates) {
+      if (isContained(candidate, ind)) {
+        contained.add(candidate);
+      }
+    }
+    return contained;
+  }
+
+  private boolean isContained(final InclusionDependency toCheck, final InclusionDependency parent) {
```

Comment:
Can't we just convert both inds to `Set<ColumIdentifier>` and the check if `setA.containsAll(setB)`? Might make this method a lot shorter.

Reply:
This is exactly what I originally started with until I added more test case values ;)
Obviously people are inclined to believe this works, probably because it is a weaker criterion. I assume the check would look similar to this one:

    lhs(parent).containsAll(lhs(toCheck)) &&
      rhs(parent).containsAll(rhs(toCheck))

However, `AB in CD` exactly says that the column of index *x* of the LHS is contained in the column at index *x* in the RHS. Using set operations destroys this connection. I encourage you to look at the test case values of the unit test.  
I was also thinking if there is any good in sorting the columns of LHS, but this would imply changing positions on RHS in the exact same way, probably without added benefit.

Reply:
Ah, got you. Good catch!